torch/csrc/autograd/python_function.cpp

```diff
@@ -210,6 +245,23 @@ auto PyNode::name() const -> std::string {
 }
 
 void PyNode::compiled_args(CompiledNodeArgs& args) {
+  pybind11::gil_scoped_acquire gil;
+
+  auto f = (THPFunction*)obj;
+
+  // update global compiled autograd key before fetching it
+  PyObject* saved_variables = unpack_saved_variables(
+        f, [](const Variable& var) { return THPVariable_Wrap(var); });
+  static PyObject* update_saved_tensors_shape =
+      PyUnicode_InternFromString("update_saved_tensors_shape");
+  THPObjectPtr update_result(PyObject_CallMethodOneArg(
+    args.get_py_compiler(),
+    update_saved_tensors_shape,
+    saved_variables));
```

Comment:
I don't think this is needed.  Don't we already include saved tensors in the shadow graph key?

Reply:
removed, we already collect saved tensors via TensorArgs::id